### ISSUE TITLE
IntegerRangeRandomizer does not generate random values in the given range

### ISSUE THREAD

**Issue by ysimhadri** [benas/random-beans#241]
Thank you for taking time to open this issue :smile:

Please provide as much context as possible to help us fixing the issue (at least random-beans version you are using).

* If you are reporting a bug, the best way is to provide a failing test.
* If you are requesting a feature, don't hesitate to explain in detail your suggestion with code examples.
* If you have a question, please first check if there is no (closed) issue about it. You may ask your question on the [Gitter channel](https://gitter.im/benas/random-beans) of the project.

Many thanks upfront!

**Comment by benas** [comment]
Hi,

> Can the EnhancedRandomBuilder only genearate positive values for integer datatypes

Here is how to to that:

```java
EnhancedRandom enhancedRandom = EnhancedRandomBuilder.aNewEnhancedRandomBuilder()
                .randomize(Integer.class, new IntegerRangeRandomizer(0, 100))
                .build();
```

Kind regards
Mahmoud

**Comment by ysimhadri** [comment]
I am getting back negative values.Drilled  down the code.
**FieldPopulator**

void populateField(final Object target, final Field field, final PopulatorContext context) throws IllegalAccessException {
        Randomizer<?> randomizer = randomizerProvider.getRandomizerByField(field);
        if (randomizer instanceof SkipRandomizer) {
            return;
        }
        context.pushStackItem(new PopulatorContextStackItem(target, field));
        if(!context.isExceedRandomizationDepth()) {
            Object value;
            if (randomizer != null) {
                **value = randomizer.getRandomValue(); IntegerRandomizer is showing up instead of IntegerRangeRandomizer**
            } else {
                value = generateRandomValue(field, context);
            }
            setProperty(target, field, value);
        }
        context.popStackItem();
    }

**Comment by benas** [comment]
Hi,

Thank you for reporting this issue! Indeed, this is a bug in random beans.
We'll fix it asap and keep you informed.

Kind regards
Mahmoud

**Comment by benas** [comment]
Hi,

I've deployed version 3.6.0-SNAPSHOT to maven central with a fix for this issue.
Could you please give it a try and tell me if this is ok for you?

Kind regards
Mahmoud

**Comment by benas** [comment]
Any update on this?

**Comment by benas** [comment]
No updates, so I'm closing this issue.
The fix will be part of v3.6

@ysimhadri In case of any problem, you can reopen this issue and it will be addressed in next version

**Comment by driversti** [comment]
Hi. The bug still exists in 3.7.0 for Long
